datatypes/src/raster/signed_grid_dimension.rs

```diff
@@ -0,0 +1,754 @@
+use std::{
+    fmt::Debug,
+    ops::{Add, Div, Mul, Rem, Sub},
+};
+
+use crate::error;
+use crate::util::Result;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+
+use super::{Dim, Dim1D, Dim2D, Dim3D, GridDimension, GridIdx2D, GridIdx3D, GridIndex};
+
+pub type SignedIdx = isize;
+
+pub type SignedGridIdx1D = Dim<[SignedIdx; 1]>;
+pub type SignedGridIdx2D = Dim<[SignedIdx; 2]>;
+pub type SignedGridIdx3D = Dim<[SignedIdx; 3]>;
+
+pub type OffsetDim1D = OffsetDim<[usize; 1], [isize; 1]>;
+pub type OffsetDim2D = OffsetDim<[usize; 2], [isize; 2]>;
+pub type OffsetDim3D = OffsetDim<[usize; 3], [isize; 3]>;
+
+/// This trait allows to address cells in a grid relative to a origin which differs from the upper left pixel.
+/// An example for this is a grid wich is anchored at a geographical coordinate (0,0) but the upper left pixel is at (-180, 90).
+pub trait OffsetDimension: Sized {
```

Comment:
why is this a custom trait? Shouldn't it be a property of a Dimension/Index whether it is relative or not?

Reply:
i think it is nice to have a simple GridDimension where you know everything is zero based. But we can merge it i don't care.

Reply:
You can now have a GridArray  with a shape  or a bounding box. they both implement the required traits and support all the needed functionality.